### PR TITLE
fuzz: fix false positives triggered by H1 codec fuzzer.

### PR DESCRIPTION
**Pull request by htuch**
This PR addresses oss-fuzz issues https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=13081.

This false positive (heap-use-after-free) resulted from a
misunderstanding around how we handle request/response encode/decoder
ownership.

As a bonus, fixed stream ID in corpus; it represents an index into the
list of created streams, not the HTTP/2 level stream ID.

Risk level: Low
Testing: Corpus entry added, validated under ASAN.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: test/common/http/codec_impl_fuzz_test.cc [hunk 1]
```diff
@@ -80,9 +80,11 @@ class HttpStream : public LinkedObject<HttpStream> {
   // course, it's the codecs must be robust to wire-level violations. We
   // explore these violations via MutateAction and SwapAction at the connection
   // buffer level.
-  enum class StreamState { PendingHeaders, PendingDataOrTrailers, Closed };
+  enum class StreamState : int { PendingHeaders, PendingDataOrTrailers, Closed };
```
Comment (mattklein123) [inline test/common/http/codec_impl_fuzz_test.cc]: Why the int?

File: test/common/http/codec_impl_fuzz_test.cc [hunk 2]
```diff
@@ -119,27 +121,28 @@ class HttpStream : public LinkedObject<HttpStream> {
       resetStream();
     }));
     ON_CALL(request_.decoder_, decodeHeaders_(_, true)).WillByDefault(InvokeWithoutArgs([this] {
+      // The HTTP/1 codec needs this to cleanup any latent stream resources.
+      response_.encoder_->getStream().resetStream(StreamResetReason::LocalReset);
       request_.closeRemote();
     }));
     ON_CALL(request_.decoder_, decodeData(_, true)).WillByDefault(InvokeWithoutArgs([this] {
+      // The HTTP/1 codec needs this to cleanup any latent stream resources.
+      ENVOY_LOG_MISC(debug, "HTD decodeData 2");
```
Comment (mattklein123) [inline test/common/http/codec_impl_fuzz_test.cc]: delete or make more intuitive?

File: test/common/http/codec_impl_fuzz_test.cc (hunk 1 above)
Comment (htuch) [reply test/common/http/codec_impl_fuzz_test.cc]: To allow it to be converted to an int in the debug log statements.

Comment (mattklein123) [reply test/common/http/codec_impl_fuzz_test.cc]: We do this elsewhere without doing this using `enumToInt` for example, but nbd.

**Pull request merged**